torch/lib/c10d/test/TCPStoreTest.cpp

```diff
@@ -1,26 +1,36 @@
 #include <c10d/test/StoreTestCommon.hpp>
 
+#include <gtest/gtest.h>
 #include <cstdlib>
 #include <iostream>
 #include <thread>
 
 #include <c10d/PrefixStore.hpp>
 #include <c10d/TCPStore.hpp>
 
-void testHelper(const std::string& prefix = "") {
+// Different ports for different tests.
+void testHelper(int port, const std::string& prefix = "") {
   const auto numThreads = 16;
   const auto numWorkers = numThreads + 1;
+
+  std::unique_ptr<c10d::TCPStore> serverTCPStore;
+  std::unique_ptr<c10d::PrefixStore> serverStore;
   // server store
-  c10d::TCPStore serverTCPStore("127.0.0.1", 29500, numWorkers, true);
-  c10d::PrefixStore serverStore(prefix, serverTCPStore);
+  auto serverThread =
+      std::thread([&serverTCPStore, &serverStore, &prefix, &numWorkers, &port] {
+        serverTCPStore = std::make_unique<c10d::TCPStore>(
```

Comment:
Should we also pass in a timeout to the constructor? That way if TCPStore ever blocks in `waitForWorkers` we can terminate quickly with an error. A good default could be ~10 seconds

Reply:
Went with 30s to be conservative.